File: typescript/js/two_sum.js
"use strict";
class TwoSum {
    constructor(nums, target) {
        this.nums = nums;
        this.target = target;
        this.nums = nums;
        this.target = target;
    }
    twoSum() {
        for (let i = 0; i < this.nums.length; i++) {
            for (let j = i + 1; j < this.nums.length; j++) {
                if (this.nums[j] === this.target - this.nums[i])
                    return [i, j];
            }
        }
    }
}
const sumObj = new TwoSum([4, 5, 6], 10);
console.log(sumObj.twoSum());

File: python/Easy/two_sum.py
'''
Given an array of integers nums and an integer target, return the indices i and j such that nums[i] + nums[j] == target and i != j.

You may assume that every input has exactly one pair of indices i and j that satisfy the condition.

Return the answer with the smaller index first.
'''
class TwoSum:
    def __init__(self, nums, target):
        self.nums = nums
        self.target = target

    def two_sum(self):
        for i in range(len(self.nums)):
            for j in range(i+1, len(self.nums)):
                if self.nums[j] == self.target - self.nums[i]:
                    return [i, j]


sumObj = TwoSum([4, 5, 6], 10)
print(sumObj.two_sum())
